tracer/src/Datadog.Trace/ClrProfiler/CallTarget/Handlers/IntegrationMapper.cs

```diff
@@ -770,6 +809,12 @@ private static void WriteCreateNewProxyInstance(ILGenerator ilWriter, Type proxy
             return (TTo)returnValue.Instance;
         }
 
+        private static async Task<TTo> UnwrapTaskReturnValue<TFrom, TTo>(Task<TFrom> returnValue, bool preserveContext)
```

Comment:
✨ Magic ✨

Reply:
Is `TFrom` guaranteed to be an `IDuckType`? 🤔 e.g. if we return `Task<string>` in the async method, then `TFrom` is `string`?

Reply:
Yes, this method only gets called when we want to `unwrap` the original instance from a ducktype proxy.